packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1090,11 +1102,44 @@ class PopupMenuButton<T> extends StatefulWidget {
   PopupMenuButtonState<T> createState() => PopupMenuButtonState<T>();
 }
 
+/// Signature for the callback used by [showMenu] to obtain the position of the
+/// [PopupMenuButton].
+///
+/// Used by [showMenu].
+typedef PopupMenuButtonPositionCallback = RelativeRect Function();
+
 /// The [State] for a [PopupMenuButton].
 ///
 /// See [showButtonMenu] for a way to programmatically open the popup menu
 /// of your button state.
 class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
+  GlobalKey _menuGlobalKey = GlobalKey();
+  RelativeRect? _buttonPosition;
```

Comment:
Could this just be a `late`? It seems like `_buttonPosition` can only be accessed once it's been initialized: 
```suggestion
  late RelativeRect _buttonPosition;
```

Reply:
late-type is a final-type, right?

Reply:
It believe it's not -- The variable can be changed to point to a different instance of the same type, just not to a null value.